site/app/controllers/HomePageController.php

```diff
@@ -275,7 +275,10 @@ public function createCoursePage() {
                 $this->core->getQueries()->getAllTerms(),
                 $this->core->getUser()->getAccessLevel() === User::LEVEL_SUPERUSER,
                 $this->core->getCsrfToken(),
-                $this->core->getQueries()->getCourseForUserId($this->core->getUser()->getId())
+                array_merge(
```

Comment:
why do we we hav an array merge?  why do we have a function call that gets the course codes only for archived courses?  wouldn't it be better to change the function to be more useful?